display_list/display_list_builder.cc

```diff
@@ -179,10 +179,38 @@ void DisplayListBuilder::onSetColorSource(const DlColorSource* source) {
     }
   }
 }
-void DisplayListBuilder::onSetImageFilter(sk_sp<SkImageFilter> filter) {
-  (current_image_filter_ = filter)  //
-      ? Push<SetImageFilterOp>(0, 0, std::move(filter))
-      : Push<ClearImageFilterOp>(0, 0);
+void DisplayListBuilder::onSetImageFilter(const DlImageFilter* filter) {
+  if (filter == nullptr) {
+    current_image_filter_ = nullptr;
+    Push<ClearImageFilterOp>(0, 0);
+  } else {
+    current_image_filter_ = filter->shared();
+    switch (filter->type()) {
+      case DlImageFilterType::kBlur: {
+        const DlBlurImageFilter* blur_filter = filter->asBlur();
+        FML_DCHECK(blur_filter);
+        void* pod = Push<SetPodImageFilterOp>(blur_filter->size(), 0);
+        new (pod) DlBlurImageFilter(blur_filter);
+        break;
+      }
+      case DlImageFilterType::kMatrix: {
+        const DlMatrixImageFilter* matrix_filter = filter->asMatrix();
+        FML_DCHECK(matrix_filter);
+        void* pod = Push<SetPodImageFilterOp>(matrix_filter->size(), 0);
+        new (pod) DlMatrixImageFilter(matrix_filter);
+        break;
+      }
+      case DlImageFilterType::kComposeFilter:
```

Comment:
There ought to be a `break` here right?

Reply:
The fall-through is intentional. Both compose and colorfilter need to be instantiated as the "SharedImageFilter" op because they both contain shared_ptr's in their bodies. I have an issue filed to address that as mentioned in the description above: https://github.com/flutter/flutter/issues/100434